tests/baselines/reference/moduleExportAssignment7.errors.txt

```diff
@@ -0,0 +1,69 @@
+tests/cases/conformance/salsa/index.ts(2,24): error TS2694: Namespace '"tests/cases/conformance/salsa/mod".export=' has no exported member 'Thing'.
+tests/cases/conformance/salsa/index.ts(3,24): error TS2694: Namespace '"tests/cases/conformance/salsa/mod".export=' has no exported member 'AnotherThing'.
+tests/cases/conformance/salsa/index.ts(4,24): error TS2694: Namespace '"tests/cases/conformance/salsa/mod".export=' has no exported member 'foo'.
+tests/cases/conformance/salsa/index.ts(5,24): error TS2694: Namespace '"tests/cases/conformance/salsa/mod".export=' has no exported member 'qux'.
+tests/cases/conformance/salsa/index.ts(6,24): error TS2694: Namespace '"tests/cases/conformance/salsa/mod".export=' has no exported member 'baz'.
+tests/cases/conformance/salsa/index.ts(18,31): error TS2694: Namespace '"tests/cases/conformance/salsa/mod".export=' has no exported member 'buz'.
+
+
+==== tests/cases/conformance/salsa/mod.js (0 errors) ====
+    class Thing  { x = 1 }
+    class AnotherThing { y = 2  }
+    function foo() { return 3 }
+    function bar() { return 4 }
+    /** @typedef {() => number} buz */
+    module.exports = {
+        Thing,
+        AnotherThing,
+        foo,
+        qux: bar,
+        baz() { return 5 }
+    }
+==== tests/cases/conformance/salsa/main.js (0 errors) ====
+    /**
+     * @param {import("./mod").Thing} a
+     * @param {import("./mod").AnotherThing} b
+     * @param {import("./mod").foo} c
+     * @param {import("./mod").qux} d
+     * @param {import("./mod").baz} e
+     * @param {import("./mod").buz} f
+     */
+    function test(a, b, c, d, e, f) {
+        return a.x + b.y + c() + d() + e() + f()
+    }
+    
+==== tests/cases/conformance/salsa/index.ts (6 errors) ====
+    function types(
+        a: import('./mod').Thing,
+                           ~~~~~
+!!! error TS2694: Namespace '"tests/cases/conformance/salsa/mod".export=' has no exported member 'Thing'.
+        b: import('./mod').AnotherThing,
+                           ~~~~~~~~~~~~
+!!! error TS2694: Namespace '"tests/cases/conformance/salsa/mod".export=' has no exported member 'AnotherThing'.
+        c: import('./mod').foo,
+                           ~~~
+!!! error TS2694: Namespace '"tests/cases/conformance/salsa/mod".export=' has no exported member 'foo'.
+        d: import('./mod').qux,
+                           ~~~
+!!! error TS2694: Namespace '"tests/cases/conformance/salsa/mod".export=' has no exported member 'qux'.
+        e: import('./mod').baz,
+                           ~~~
+!!! error TS2694: Namespace '"tests/cases/conformance/salsa/mod".export=' has no exported member 'baz'.
+        f: import('./mod').buz,
+    ) {
+        return a.x + b.y + c() + d() + e() + f()
+    }
+    
+    function values(
+        a: typeof import('./mod').Thing,

```

Comment:
Similarly, this sure looks like it would be referring to the constructor type of `Thing`, but it’s the instance type here.